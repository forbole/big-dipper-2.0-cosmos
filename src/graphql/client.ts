import WebSocket from 'isomorphic-ws';
import {
  ApolloClient,
  InMemoryCache,
  split,
  HttpLink,
  ApolloLink,
  concat,
} from '@apollo/client';
import {
  getMainDefinition,
} from '@apollo/client/utilities';
import { WebSocketLink } from '@apollo/client/link/ws';

import { useMemo } from 'react';

const defaultOptions:any = {
  watchQuery: {
    fetchPolicy: 'no-cache',
    errorPolicy: 'all',
  },
  query: {
    fetchPolicy: 'no-cache',
    errorPolicy: 'all',
  },
};

let apolloClient;

const httpLink = new HttpLink({
  uri: process.env.NEXT_PUBLIC_GRAPHQL_URL,
});

const wsLink = new WebSocketLink({
  uri: process.env.NEXT_PUBLIC_GRAPHQL_WS ?? 'wss://localhost:3000',
  options: {
    reconnect: true,
  },
  webSocketImpl: WebSocket,
});

const link = typeof window !== 'undefined' ? split(
  ({ query }) => {
    const {
      kind, operation,
    }:any = getMainDefinition(query);
    return kind === 'OperationDefinition' && operation === 'subscription';
  },
  wsLink,
  httpLink,
) : httpLink;

const authMiddleware = new ApolloLink((operation, forward) => {
  operation.setContext({
    headers: {
    },
  });

  return forward(operation);
});

function createApolloClient() {
  const client = new ApolloClient({
    ssrMode: typeof window === 'undefined',
    link: concat(authMiddleware, link),
    cache: new InMemoryCache({
    }),
  });

  client.defaultOptions = defaultOptions;

  return client;
}

export function initializeApollo(initialState = null) {
  // eslint-disable-next-line
  const _apolloClient = apolloClient ?? createApolloClient();

  // If your page has Next.js data fetching methods that use Apollo Client, the initial state
  // gets hydrated here
  if (initialState) {
    // Get existing cache, loaded during client side data fetching
    const existingCache = _apolloClient.extract();
    // Restore the cache using the data passed from getStaticProps/getServerSideProps
    // combined with the existing cached data
    _apolloClient.cache.restore({
      ...existingCache, ...initialState,
    });
  }
  // For SSG and SSR always create a new Apollo Client
  if (typeof window === 'undefined') return _apolloClient;
  // Create the Apollo Client once in the client
  if (!apolloClient) apolloClient = _apolloClient;

  return _apolloClient;
}

export function useApollo(initialState) {
  const store = useMemo(() => initializeApollo(initialState), [initialState]);
  return store;
}
