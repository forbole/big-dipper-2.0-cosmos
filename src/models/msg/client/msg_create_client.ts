import { Categories } from '../types';

class MsgCreateClient {
    public category: Categories;
    public type: string;
    public signer: string;
    public chainId: string;
    public json: any;

    constructor(payload: any) {
      this.category = 'client';
      this.type = payload.type;
      this.signer = payload.owner;
      this.chainId = payload.chainId;
      this.json = payload.json;
    }

    static fromJson(json: any) {
      return new MsgCreateClient({
        json,
        type: json['@type'],
        signer: json.signer,
        chainId: json.client_state?.chain_id,
      });
    }
}

export default MsgCreateClient;
