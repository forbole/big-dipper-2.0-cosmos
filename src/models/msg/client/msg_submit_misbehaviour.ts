import { Categories } from '../types';

class MsgSubmitMisbehaviour {
    public category: Categories;
    public type: string;
    public signer: string;
    public clientId: string;
    public misbehaviour: string;
    public json: any;

    constructor(payload: any) {
      this.category = 'client';
      this.type = payload.type;
      this.signer = payload.owner;
      this.clientId = payload.clientId;
      this.misbehaviour = payload.misbehaviour;
      this.json = payload.json;
    }

    static fromJson(json: any) {
      return new MsgSubmitMisbehaviour({
        json,
        type: json['@type'],
        signer: json.signer,
        clientId: json.client_id,
        misbehaviour: json.misbehaviour,
      });
    }
}

export default MsgSubmitMisbehaviour;
