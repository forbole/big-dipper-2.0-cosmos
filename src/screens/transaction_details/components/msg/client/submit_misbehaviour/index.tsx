import React from 'react';
import Trans from 'next-translate/Trans';
import { Typography } from '@material-ui/core';
import { Name } from '@components';
import { MsgSubmitMisbehaviour } from '@models';
import { useChainContext } from '@contexts';

const SubmitMisbehaviour = (props: {
    message: MsgSubmitMisbehaviour;
}) => {
  const { findAddress } = useChainContext();
  const { message } = props;

  const signer = findAddress(message.signer);
  const signerMoniker = signer ? signer?.moniker : message.signer;

  return (
    <Typography>
      <Trans
        i18nKey="message_contents:txSubmitMisbehaviourContent"
        components={[
          (
            <Name
              address={message.signer}
              name={signerMoniker}
            />
          ),
          <b />,
        ]}
        values={{
          clientId: message.clientId,
        }}
      />
    </Typography>
  );
};

export default SubmitMisbehaviour;
