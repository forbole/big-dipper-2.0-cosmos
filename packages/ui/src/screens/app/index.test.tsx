import MyApp from '@/screens/app';
import { ApolloClient, from, InMemoryCache } from '@apollo/client';
import type { Router } from 'next/router';
import renderer from 'react-test-renderer';

// ==================================
// mocks
// ==================================
const mockClient = new ApolloClient({ link: from([]), cache: new InMemoryCache() });

jest.mock('next-translate/useTranslation', () => () => ({
  lang: 'en',
}));
jest.mock('@/graphql/useApollo', () => () => mockClient);

const mockI18n = {
  t: (key: string) => key,
  lang: 'en',
};

jest.mock('next-translate/useTranslation', () => () => mockI18n);

jest.mock('@/screens/app/hooks', () => ({
  useApp: () => jest.fn(),
}));

// ==================================
// unit tests
// ==================================
describe('screen: _app', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <MyApp router={{} as Router} Component={() => <div id="component" />} pageProps={{}} />
    );
    const tree = component?.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
