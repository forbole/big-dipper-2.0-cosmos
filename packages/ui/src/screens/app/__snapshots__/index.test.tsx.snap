// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: _app matches snapshot 1`] = `
[
  <div
    className="Toastify"
  />,
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  min-height: 100vh;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #F8F8F8;
}

.emotion-0 a {
  color: #4092CD;
}

.emotion-1 {
  width: 275px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-2 {
  width: auto;
  height: auto;
  max-width: 100%;
  max-height: 100%;
  min-width: 100%;
  display: none;
}

.mode-dark .emotion-2 {
  display: initial;
}

.emotion-3 {
  width: auto;
  height: auto;
  max-width: 100%;
  max-height: 100%;
  min-width: 100%;
  object-fit: contain;
}

.mode-dark .emotion-3 {
  display: none;
}

.emotion-4 {
  display: grid;
  grid-template-columns: repeat(4, 65px);
  gap: 8px;
  margin: 24px 0px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-5 .MuiTypography-h1 {
  width: 100%;
  text-align: center;
  background: #FFFFFF;
  padding: 16px;
  margin-bottom: 16px;
}

.emotion-6 {
  margin: 0;
  font-size: 2rem;
  letter-spacing: 0.25px;
  font-family: fontFamily;
  font-weight: 300;
  line-height: 1.167;
}

.emotion-7 {
  margin: 0;
  font-size: 1.25rem;
  letter-spacing: 0.15px;
  font-family: fontFamily;
  font-weight: 400;
  line-height: 1.167;
}

.emotion-17 {
  margin: 0;
  font-size: 1.5rem;
  letter-spacing: 0;
  font-family: fontFamily;
  font-weight: 300;
  line-height: 1.2;
  color: #FF835B;
}

<div
    className="emotion-0"
  >
    <span
      className="emotion-1"
    >
      <img
        alt="logo"
        className="emotion-2"
        data-nimg="1"
        decoding="async"
        height={0}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="svg"
        style={
          {
            "color": "transparent",
          }
        }
        width={0}
      />
      <img
        alt="logo"
        className="emotion-3"
        data-nimg="1"
        decoding="async"
        height={0}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="svg"
        style={
          {
            "color": "transparent",
          }
        }
        width={0}
      />
    </span>
    <div
      className="emotion-4"
    >
      <div
        className="emotion-5"
      >
        <h1
          className="MuiTypography-root MuiTypography-h1 emotion-6"
        >
          0
        </h1>
        <h3
          className="MuiTypography-root MuiTypography-h3 emotion-7"
        >
          Day
        </h3>
      </div>
      <div
        className="emotion-5"
      >
        <h1
          className="MuiTypography-root MuiTypography-h1 emotion-6"
        >
          0
        </h1>
        <h3
          className="MuiTypography-root MuiTypography-h3 emotion-7"
        >
          Hour
        </h3>
      </div>
      <div
        className="emotion-5"
      >
        <h1
          className="MuiTypography-root MuiTypography-h1 emotion-6"
        >
          0
        </h1>
        <h3
          className="MuiTypography-root MuiTypography-h3 emotion-7"
        >
          Min
        </h3>
      </div>
      <div
        className="emotion-5"
      >
        <h1
          className="MuiTypography-root MuiTypography-h1 emotion-6"
        >
          0
        </h1>
        <h3
          className="MuiTypography-root MuiTypography-h3 emotion-7"
        >
          Sec
        </h3>
      </div>
    </div>
    <h2
      className="MuiTypography-root MuiTypography-h2 emotion-17"
    >
      desmos-mainnet
    </h2>
  </div>,
]
`;
